### PR TITLE
jstock: Add version 1.0.7.45

### PR DESCRIPTION
**Pull request by GotAnAccount**
Adding the JStock app scripts. ;)

### CODE REVIEW

File: bucket/jstock.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    "homepage": "https://jstock.org",
+    "description": "JStock is a free stock market software for 28 countries",
+    "license": "GPL-2.0-only",
+    "version": "1.0.7.45",
+    "extract_dir": "jstock",
+    "architecture": {
+        "64bit": {
```
Comment (Ash258) [inline bucket/jstock.json:8]: What is behind your assumption this is 64bit only?

https://github.com/yccheok/jstock/blob/ba607c28beb780936bbf23abdc61dc2b6de9f072/installer/windows/jstock.xml#L24

File: bucket/jstock.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            "java/oraclejdk",
+            "java/openjdk"
+        ]
+    },
+    "shortcuts": [
+        [
+            "jstock.bat",
+            "JStock"
+        ]
+    ]
```
Comment (Ash258) [inline bucket/jstock.json:25]:
```suggestion
    "version": "1.0.7.45",
    "description": "Stock investments tracker and manager",
    "homepage": "https://jstock.org",
    "license": "GPL-2.0-only",
    "suggest": {
        "JDK": [
            "java/oraclejdk",
            "java/openjdk"
        ]
    },
    "architecture": {
        "64bit": {
            "url": "https://github.com/yccheok/jstock/releases/download/release_1-0-7-45/jstock-1.0.7.45-bin.zip",
            "hash": "6a795a61258c3d9488b40cb449a1de9ea263624bd1a0e6c0ea73de6293e8a75b"
        }
    },
    "extract_dir": "jstock",
    "pre_install": "Set-Content \"$dir\\jstock.bat\" '@start javaw.exe -jar \"%~dp0jstock.jar\" %*' -Encoding Ascii",
    "shortcuts": [
        [
            "jstock.bat",
            "JStock"
        ]
    ],
    "checkver": {
        "github": "https://github.com/yccheok/jstock",
        "regex": "jstock-([\\d.]+)-bin\\.zip"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://github.com/yccheok/jstock/releases/download/release_$dashVersion/jstock-$version-bin.zip"
            }
        }
    }
```

**Review by Ash258 — CHANGES_REQUESTED** [review]
What is `config` directory for?

File: bucket/jstock.json (hunk 1 above)
Comment (GotAnAccount) [reply bucket/jstock.json:8]: I have not tested it on 32bit so I don't assume it works.

File: bucket/jstock.json (hunk 2 above)
Comment (GotAnAccount) [reply bucket/jstock.json:25]: I had not thought of adding the two section as you just did. nice addition

**Comment by GotAnAccount** [comment]
Thanks for the review.

**Review by GotAnAccount — COMMENTED** [review]
all good

**Comment by Ash258** [comment]
@rasa: Why do you spam PRs, which are not complete with /verify??

**Comment by rasa** [comment]
> @rasa: Why do you spam PRs, which are not complete with /verify??

To verify if there are any other issues that need to be addressed. We generally run /verify to be able to provide feedback before reviewing/approving. Are you suggesting we review/approve first, and then run /verify?

**Comment by Ash258** [comment]
The requested changes were not implemented https://github.com/lukesampson/scoop-extras/pull/5387#discussion_r549877003

**Comment by rasa** [comment]
Also, @GotAnAccount said "all good" so it appeared they were ready for it to be reviewed.

**Comment by rasa** [comment]
> The requested changes were not implemented [#5387 (comment)](https://github.com/lukesampson/scoop-extras/pull/5387#discussion_r549877003)

The requested changes appear to be regarding formatting. Are we now requiring users to follow a particular format, that otherwise passes verification? If so, it would be helpful to our users to point out this requirement in our contributing docs. Also, perhaps `/verify` can fail if the formatting fails to pass muster. And perhaps we can add a `/reformat` flag to allow us to reformst as needed?

File: bucket/jstock.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+    "version": "1.0.7.55",
+    "description": "Free stock market software for 28 countries",
+    "homepage": "https://jstock.org",
+    "license": "GPL-2.0-only",
+    "extract_dir": "jstock",
+    "architecture": {
+        "64bit": {
```
Comment (rashil2000) [inline bucket/jstock.json:8]: @issaclin32 arch block should be removed

**Review by rashil2000 — CHANGES_REQUESTED** [review]
Should the `config` folder be persisted?

**Comment by issaclin32** [comment]
> Should the `config` folder be persisted?

Doesn't look like so.
The config is stored at `$Env:UserProfile\.jstock`, and the only thing I found in `config` folder is this:

**File: commons-logging.properties**
```
# Sample ResourceBundle properties file
#org.apache.commons.logging.Log=org.apache.commons.logging.impl.NoOpLog
```

**Review by rashil2000 — APPROVED** [review]

**Pull request merged**